autoload/go/fmt.vim

```diff
@@ -52,13 +52,25 @@ endif
 "  this and have VimL experience, please look at the function for
 "  improvements, patches are welcome :)
 function! go#fmt#Format(withGoimport)
-    " save cursor position, folds and many other things
-    let l:curw = {}
-    try
-        mkview!
-    catch
+    if g:go_fmt_experimental == 1
+        " Using winsaveview to save/restore cursor state has the problem of
+        " closing folds on save:
+        "   https://github.com/fatih/vim-go/issues/502
+        " One fix is to use mkview instead. Unfortunately, this sometimes causes
+        " other bad side effects:
+        "   https://github.com/fatih/vim-go/issues/728
+        " and still closes all folds if foldlevel>0:
+        "   https://github.com/fatih/vim-go/issues/732
+        let l:curw = {}
+        try
+            mkview!
+        catch
+            let l:curw=winsaveview()
+        endtry
+    else
```

Comment:
@fatih Sorry, I've read your message several times and I don't see the problem. What's wrong with the `else` branch?

You wrote

> This change is now putting it behind a flag.

but it's not behind a flag; it's in the `else` branch where `g:go_fmt_experimental != 1`.